### DISCUSSION TITLE
How to use `dtype` in `sqrt`?

### DISCUSSION THREAD

**Discussion by OverLordGoldDragon** [jax-ml/jax#13389] [Q&A] [answered]
```python
import jax.numpy as jnp

x = jnp.array([1.])
jnp.sqrt(x, dtype=jnp.float32)
```
```python
TypeError: Argument '<class 'jax.numpy.float32'>' of type <class 'jax._src.numpy.lax_numpy._ScalarMeta'> 
is not a valid JAX type.
```
```python
jnp.sqrt(x, dtype=x.dtype)
```
```python
TypeError: Argument 'float32' of type <class 'numpy.dtype[float32]'> is not a valid JAX type.
```
So what's a valid `dtype`? My intent is to mimic `np.sqrt(dtype=)` functionality, mainly to type-cast, and optionally to accept scalar inputs.

Windows 10, CUDA 11.3, Anaconda, `pip install "jax[cpu]===0.3.14" -f https://whls.blob.core.windows.net/unstable/index.html --use-deprecated legacy-resolver`.

**Comment by jakevdp** [comment, marked as answer]
The error could be improved here, but if you look at the [function documentation](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.sqrt.html) you'll see that `jnp.sqrt` only takes a single argument. If you need to specify the dtype in JAX, you can get equivalent functionality by doing
```python
jnp.sqrt(x.astype(dtype))
```
